File: finanze/domain/use_cases/connect_crypto_wallet.py
import abc

from domain.crypto import (
    ConnectCryptoWallet as ConnectCryptoWalletRequest,
)
from domain.crypto import (
    CryptoWalletConnectionResult,
)


class ConnectCryptoWallet(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def execute(self, data: ConnectCryptoWalletRequest) -> CryptoWalletConnectionResult:
        raise NotImplementedError

File: mobile/src/domain/usecases/core/connectCryptoWallet.ts
import {
  ConnectCryptoWallet as ConnectCryptoWalletRequest,
  CryptoWalletConnectionResult,
} from "@/domain"

export interface ConnectCryptoWallet {
  execute(
    data: ConnectCryptoWalletRequest,
  ): Promise<CryptoWalletConnectionResult>
}
